examples/systemcalls/systemcalls.c

```diff
@@ -49,15 +74,39 @@ bool do_exec(int count, ...)
     // and may be removed
     command[count] = command[count];
 
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
+    if (command[0][0] != '/')
+    {
+        syslog(LOG_ERR, "The file path is not an absolute one.");
+        return false;
+    }
+
+    pid = fork();
+    if (pid == -1)
+    {
+        syslog(LOG_ERR, "fork() failed with an error: %s", strerror(errno));
+        return false;
```

Comment:
@krishnasuhagiya29 what happens to `va_end` cleanup in each of these code paths?